src/support.js

```diff
@@ -84,41 +84,102 @@ var asm2wasmImports = { // special asm2wasm imports
 };
 
 #if RELOCATABLE
+// fetchBinary fetches binaray data @ url. (async)
+function fetchBinary(url) {
+  return fetch(url, { credentials: 'same-origin' }).then(function(response) {
+    if (!response['ok']) {
+      throw "failed to load binary file at '" + url + "'";
+    }
+    return response['arrayBuffer']();
+  }).then(function(buffer) {
+    return new Uint8Array(buffer);
+  });
+}
 
-function loadDynamicLibrary(lib) {
-  var libModule;
+// loadDynamicLibrary loads dynamic library @ lib URL / path.
+//
+// Several flags affect the loading:
+//
+// - if flags.loadAsync=true, the loading is performed asynchronously and
+//   loadDynamicLibrary returns corresponding promise.
+function loadDynamicLibrary(lib, flags) {
+  if (flags === undefined) {
```

Comment:
can do `flags = flags || {}` which is a pattern we use elsewhere

Reply:
sure, thanks for the advice.